test/e2e/service.go

```diff
@@ -232,28 +231,131 @@ var _ = Describe("Services", func() {
 		}()
 
 		name1 := "test1"
-		addEndpointPodOrFail(c, ns, name1, labels)
+		addEndpointPodOrFail(c, ns, name1, labels, []api.ContainerPort{{ContainerPort: 80}})
 		names = append(names, name1)
 
-		validateEndpointsOrFail(c, ns, serviceName, expectedPort, names)
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name1: {80}})
 
 		name2 := "test2"
-		addEndpointPodOrFail(c, ns, name2, labels)
+		addEndpointPodOrFail(c, ns, name2, labels, []api.ContainerPort{{ContainerPort: 80}})
 		names = append(names, name2)
 
-		validateEndpointsOrFail(c, ns, serviceName, expectedPort, names)
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name1: {80}, name2: {80}})
 
 		err = c.Pods(ns).Delete(name1, nil)
 		Expect(err).NotTo(HaveOccurred())
 		names = []string{name2}
 
-		validateEndpointsOrFail(c, ns, serviceName, expectedPort, names)
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name2: {80}})
 
 		err = c.Pods(ns).Delete(name2, nil)
 		Expect(err).NotTo(HaveOccurred())
 		names = []string{}
 
-		validateEndpointsOrFail(c, ns, serviceName, expectedPort, names)
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{})
+
+		// We deferred Gingko pieces that may Fail, we aren't done.
+		defer func() {
+			close(done)
+		}()
+	}, 240.0)
+
+	It("should serve multiport endpoints from pods", func(done Done) {
+		// repacking functionality is intentionally not tested here - it's better to test it in an integration test.
+		serviceName := "multi-endpoint-test"
+		ns := api.NamespaceDefault
```

Comment:
Ok. I'm rewriting also other tests:
	It("should provide DNS for the cluster", func() {    - rewritten to use createTestingNS
	It("should serve a basic endpoint from pods", func(done Done) - doesn't use namespaces
	It("should serve a basic endpoint from pods", func(done Done)  - rewritten
	It("should serve multiport endpoints from pods", func(done Done) - the new test (rewritten)
	It("should be able to create a functioning external load balancer", func() - uses custom namespaces
	It("should correctly serve identically named services in different namespaces on different external IP addresses", func() - uses custom namespaces

So currently some tests will not use the namespace creates in BeforeEach. I hope this is ok.





	It("should provide RW and RO services", func() {